src/tesla-vehicle-manager.ts

```diff
@@ -1,29 +1,32 @@
-import { TeslaVehicleCommand } from './tesla-vehicle';
-import { TeslaVehicleClient, TeslaVehicleDetails } from './tesla-vehicle';
-import axios, { AxiosInstance, AxiosRequestConfig } from 'axios';
-import { OAuthLoginDetails } from './oAuthClient';
-import { saveToken } from './config';
+import { TeslaVehicle, TeslaVehicleDetails, getAxiosInstance, TeslaVehicleCommand } from './tesla-vehicle';
+import { AxiosInstance } from 'axios';
+
+export const command = TeslaVehicleCommand;
 
 /**
  * Logs you in to your Tesla account, and does things to your vehicles.
  */
-export class TeslaVehicleManager implements TeslaVehicleClient {
-	static fragments = {
-		login: 'oauth/token?grant_type=password',
-		refresh: 'oauth/token?grant_type=refresh_token'
-	};
-	static oAuthDetails = {
-		clientId: '81527cff06843c8634fdc09e8ac0abefb46ac849f38fe1e431c2ef2106796384',
-		clientSecret: 'c7257eb71a564034f9419ee651c7d0e5f7aa6bfbd18bafb5c5c033b093bb2fa3'
-	};
+export class TeslaVehicleManager implements TeslaVehicle {
 	private _axiosClient: AxiosInstance;
 	private _vehicleCache: Array<TeslaVehicleDetails> = [];
-	constructor() {
-		this._axiosClient = this._getAxiosInstance();
+
+	static async getVehicle(token: string): Promise<TeslaVehicle> {
+		do {
+			try {
+				return await new TeslaVehicleManager(token)._wakeVehicle();
+			} catch (e) {}
+		} while (true);
 	}
-	async getBatteryLevels() {
-		return Promise.all((await this.getVehicles()).map(({ id }) => this._getBatteryLevel(id)));
+
+	private constructor(token: string) {
+		this._axiosClient = getAxiosInstance(token);
 	}
+
+	public async getBatteryLevel() {
+		return (await this._axiosClient.get(`/${await this._getId()}/data_request/charge_state`)).data.response
+			.battery_level;
+	}
+
 	async getVehicles(): Promise<Array<TeslaVehicleDetails>> {
```

Comment:
the vehicle details should not be exposed. always encapsulate the intent that interacts with data with an interface.